graphql-kotlin-federation/src/test/kotlin/com/expediagroup/graphql/federation/validation/ValidateRequiresDirectiveKtTest.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.federation.validation
+
+import graphql.Scalars.GraphQLString
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLObjectType
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+
+class ValidateRequiresDirectiveKtTest {
```

Comment:
those tests are not as explicit as previous ones

- checks for valid schema
```
    type SimpleRequires @extends @key(fields : "id") {
      description: String!
      id: String! @external
      shippingCost: String! @requires(fields : "weight")
      weight: Float! @external
    }
```
- `@requires` references local field
```
    type RequiresLocalField @extends @key(fields : "id") {
      description: String!
      id: String! @external
      shippingCost: String! @requires(fields : "weight")
      weight: Float!
    }
```
- non-existent field referenced
```
    type RequiresNonExistentField @extends @key(fields : "id") {
      description: String!
      id: String! @external
      shippingCost: String! @requires(fields : "weight")
      weight: Float! @external
    }
```
- base type uses `@requires`
```
    type RequiresOnLocalType @key(fields : "id") {
      description: String!
      id: String!
      shippingCost: String! @requires(fields : "weight")
      weight: Float!
    }
```

Reply:
while tests below now have proper `@requires` cases they still do not have valid types, e.g.`@extends` without `@key`

Reply:
@dariuszkuc I was only trying to cover all the logic in `validateRequiresDirective` which is really just one conditional. 

After that I test all the other method in their own unit tests so I didn't see the need to tests all possible edge cases through this method.

I can add them though if we want to make these some more integration level tests.

Reply:
@smyrick without doing the proper setup those unit tests are just testing stuff in order to bump the code coverage which seems wrong to me

Reply:
Checking the key directive on the parent type doesn't happen anywhere underneath `validateRequiresDirective` as it only accepts the parent type name for logging purposes.

Reply:
I have moved the valid tests cases to run from `toFederatedSchema` in their own packages. We can only have passing schemas here though because as soon as we include an invalid type marked with `@extend` all test cases will fail